src/edits.rs

```diff
@@ -90,7 +90,16 @@ where
     }
     // Emit any remaining plus lines
     for plus_line in &plus_lines[plus_index..] {
-        annotated_plus_lines.push(vec![(noop_insertions[plus_index], plus_line)]);
+        // Split the line if it contains extra whitespace at the end preceded with non whitespace
+        let line = plus_line.trim_end_matches(&[' ', '\t', '\n']);
```

Comment:
\n won't match anything if this is already line-split, did you mean \r ?

Reply:
The line still contain the `\n` at the end and it need to be taken care of